File: cases/taint/python/exception-catch-positive/exception_catch.py
class FlowException(Exception):
    pass


def dfb_source():  # DFB-SOURCE: exception-catch-input
    return "tainted"


def dfb_sink(value):  # DFB-SINK: exception-catch-sink
    pass


def run():
    try:
        flow = FlowException()
        flow.value = dfb_source()
        raise flow  # DFB-WITNESS: exception-catch-throw
    except FlowException as caught:
        dfb_sink(caught.value)

File: cases/taint/ruby/exception-catch-positive/exception_catch.rb
class FlowError < StandardError
  attr_accessor :value
end

def dfb_source # DFB-SOURCE: exception-catch-input
  "tainted"
end

def dfb_sink(value) # DFB-SINK: exception-catch-sink
end

def run
  begin
    flow = FlowError.new
    flow.value = dfb_source
    raise flow # DFB-WITNESS: exception-catch-throw
  rescue FlowError => caught
    dfb_sink(caught.value)
  end
end
